library/ssl_tls.c

```diff
@@ -978,25 +989,25 @@ void ssl_calc_verify_ssl( mbedtls_ssl_context *ssl, unsigned char hash[36] )
     memset( pad_1, 0x36, 48 );
     memset( pad_2, 0x5C, 48 );
 
-    mbedtls_md5_update( &md5, ssl->session_negotiate->master, 48 );
-    mbedtls_md5_update( &md5, pad_1, 48 );
-    mbedtls_md5_finish( &md5, hash );
+    mbedtls_md5_update_ext( &md5, ssl->session_negotiate->master, 48 );
```

Comment:
As we discussed, handling the return values here would be best done after the message processing stack is refactored. Until then we shouldn't loose track of this issue and document it. Therefore I suggest
- raising a ticket for this issue
- mentioning in the relevant Changelog entry that the `ssl_tls.c` module is yet to be updated